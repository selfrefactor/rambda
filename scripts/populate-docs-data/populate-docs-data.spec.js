import { populateDocsData } from './populate-docs-data.js'

test('happy', async () => {
  await expect(populateDocsData()).resolves.toBeObject()
})
