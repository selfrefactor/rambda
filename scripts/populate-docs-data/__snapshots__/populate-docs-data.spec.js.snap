// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`happy 1`] = `
Object {
  "F": Object {
    "typing": "F(): boolean",
  },
  "T": Object {
    "typing": "T(): boolean",
  },
  "add": Object {
    "example": "\`\`\`javascript
R.add(2, 3) // =>  5
\`\`\`",
    "explanation": "It adds \`a\` and \`b\`. It doesn't work with strings, as the inputs are parsed to numbers before calculation.",
    "note": "It doesn't work with strings",
    "typing": "add(a: number, b: number): number",
  },
  "adjust": Object {
    "example": "\`\`\`javascript
R.adjust(
  0,
  a => a + 1,
  [0, 100]
) // => [1, 100]
\`\`\`",
    "explanation": "It replaces \`index\` in array \`list\` with the result of \`replaceFn(arr[i])\`.",
    "failedRamdaSpecs": "var R = require('../../../../dist/rambda.js');
var eq = require('./shared/eq');

describe('adjust', function() {
  it('accepts an array-like object', function() {
    function args() {
      return arguments;
    }
    eq(R.adjust(2, R.add(1), args(0, 1, 2, 3)), [0, 1, 3, 3]);
  });
});",
    "note": "",
    "typing": "adjust<T>(index: number, replaceFn: (a: T) => T, list: ReadonlyArray<T>): T[]",
  },
  "all": Object {
    "example": "\`\`\`javascript
const list = [ 0, 1, 2, 3, 4 ]
const predicate = x => x > -1

const result = R.all(fn, arr)
// => true
\`\`\`",
    "explanation": "It returns \`true\`, if all members of array \`list\` returns \`true\`, when applied as argument to \`predicate\` function.",
    "note": "",
    "typing": "all<T>(predicate: (x: T) => boolean, list: ReadonlyArray<T>): boolean",
  },
  "allPass": Object {
    "typing": "allPass<T>(predicates: ((x: T) => boolean)[]): (input: T) => boolean",
  },
  "always": Object {
    "typing": "always<T>(x: T): () => T",
  },
  "and": Object {
    "typing": "and<T extends { and?: ((...a: readonly any[]) => any)",
  },
  "any": Object {
    "typing": "any<T>(fn: (x: T, i: number) => boolean, arr: ReadonlyArray<T>): boolean",
  },
  "anyPass": Object {
    "typing": "anyPass<T>(preds: ReadonlyArray<SafePred<T>>): SafePred<T>",
  },
  "append": Object {
    "typing": "append<T>(el: T, list: ReadonlyArray<T>): T[]",
  },
  "applySpec": Object {
    "typing": "applySpec<Obj extends Record<string, (...args: readonly any[]) => any>>(
  obj: Obj
): (
    ...args: Parameters<ValueOfRecord<Obj>>
  ) => { [Key in keyof Obj]: ReturnType<Obj[Key]> }",
  },
  "assoc": Object {
    "typing": "assoc<T, U, K extends string>(prop: K, value: T, obj: U): Record<K, T> & U",
  },
  "assocPath": Object {
    "typing": "assocPath<T, U>(path: Path, val: T, obj: U): U",
  },
  "both": Object {
    "typing": "both(pred1: Pred, pred2: Pred): Pred",
  },
  "clamp": Object {
    "typing": "clamp(min: number, max: number, input: number): number",
  },
  "clone": Object {
    "typing": "clone<T>(value: T): T",
  },
  "complement": Object {
    "typing": "complement(pred: (...args: any[]) => boolean): (...args: any[]) => boolean",
  },
  "compose": Object {
    "typing": "compose<T1>(fn0: () => T1): () => T1",
  },
  "concat": Object {
    "typing": "concat<T>(x: ReadonlyArray<T>, y: ReadonlyArray<T>): T[]",
  },
  "cond": Object {
    "typing": "cond(fns: [Pred, (...a: readonly any[]) => any][]): (...a: readonly any[]) => any",
  },
  "curry": Object {
    "typing": "curry<F extends (...args: any) => any>(f: F): FToolbelt.Curry<F>",
  },
  "dec": Object {
    "typing": "dec(n: number): number",
  },
  "defaultTo": Object {
    "typing": "defaultTo<T>(defaultValue: T): (...inputArguments: (T | null | undefined)[]) => T",
  },
  "difference": Object {
    "typing": "difference<T>(list1: ReadonlyArray<T>, list2: ReadonlyArray<T>): T[]",
  },
  "dissoc": Object {
    "typing": "dissoc<T>(prop: string, obj: any): T",
  },
  "divide": Object {
    "typing": "divide(a: number, b: number): number",
  },
  "drop": Object {
    "typing": "drop<T>(howManyToDrop: number, arrOrStr: ReadonlyArray<T>): T[]",
  },
  "dropLast": Object {
    "typing": "dropLast<T>(howManyToDrop: number, arrOrStr: ReadonlyArray<T>): T[]",
  },
  "either": Object {
    "typing": "either(pred1: Pred, pred2: Pred): Pred",
  },
  "endsWith": Object {
    "typing": "endsWith(a: string, list: string): boolean",
  },
  "equals": Object {
    "typing": "equals<T>(a: T, b: T): boolean",
  },
  "filter": Object {
    "typing": "filter<T>(filterFn: FilterFunctionArray<T>): (x: T[]) => T[]",
  },
  "find": Object {
    "typing": "find<T>(findFn: (a: T) => boolean, arr: ReadonlyArray<T>): T | undefined",
  },
  "findIndex": Object {
    "typing": "findIndex<T>(findFn: (a: T) => boolean, arr: ReadonlyArray<T>): number",
  },
  "flatten": Object {
    "typing": "flatten<T>(x: ReadonlyArray<T> | ReadonlyArray<T[]> | ReadonlyArray<ReadonlyArray<T>>): T[]",
  },
  "flip": Object {
    "typing": "flip<T, U, TResult>(fn: (arg0: T, arg1: U) => TResult): (arg1: U, arg0?: T) => TResult",
  },
  "forEach": Object {
    "typing": "forEach<T>(fn: (x: T) => void, list: T[]): T[]",
  },
  "fromPairs": Object {
    "typing": "fromPairs<V>(pairs: KeyValuePair<string, V>[]): { [index: string]: V }",
  },
  "groupBy": Object {
    "typing": "groupBy<T>(fn: (a: T) => string, list: ReadonlyArray<T>): { [index: string]: T[] }",
  },
  "groupWith": Object {
    "typing": "groupWith<T>(fn: (x: T, y: T) => boolean): (list: ReadonlyArray<T>) => T[][]",
  },
  "has": Object {
    "typing": "has<T>(prop: string, obj: T): boolean",
  },
  "head": Object {
    "typing": "head<T>(arrOrStr: T[]): T | undefined",
  },
  "identical": Object {
    "typing": "identical<T>(a: T, b: T): boolean",
  },
  "identity": Object {
    "typing": "identity<T>(x: T): T",
  },
  "ifElse": Object {
    "typing": "ifElse(fn: Pred, onTrue: Arity1Fn, onFalse: Arity1Fn): Arity1Fn",
  },
  "inc": Object {
    "typing": "inc(n: number): number",
  },
  "includes": Object {
    "typing": "includes(valueToFind: string, input: ReadonlyArray<string> | string): boolean",
  },
  "indexBy": Object {
    "typing": "indexBy<T>(condition: (a: T) => string, arr: ReadonlyArray<T>): { [key: string]: T }",
  },
  "indexOf": Object {
    "typing": "indexOf<T>(target: T, arr: ReadonlyArray<T>): number",
  },
  "init": Object {
    "typing": "init<T>(arrOrStr: ReadonlyArray<T>): T[]",
  },
  "intersection": Object {
    "typing": "intersection<T>(list1: ReadonlyArray<T>, list2: ReadonlyArray<T>): T[]",
  },
  "intersperse": Object {
    "typing": "intersperse<T>(separator: T, list: ReadonlyArray<T>): T[]",
  },
  "is": Object {
    "typing": "is(xPrototype: any, x: any): boolean",
  },
  "isEmpty": Object {
    "typing": "isEmpty<T>(input: T): boolean",
  },
  "isNil": Object {
    "typing": "isNil(x: any): x is null | undefined",
  },
  "join": Object {
    "typing": "join(x: string, xs: ReadonlyArray<any>): string",
  },
  "keys": Object {
    "typing": "keys<T extends object>(x: T): (keyof T)[]",
  },
  "last": Object {
    "typing": "last<T>(arrOrStr: T[]): T | undefined",
  },
  "lastIndexOf": Object {
    "typing": "lastIndexOf<T>(x: T, arr: ReadonlyArray<T>): number",
  },
  "length": Object {
    "typing": "length<T>(list: ReadonlyArray<T>): number",
  },
  "lens": Object {
    "typing": "lens<T, U, V>(getter: (s: T) => U, setter: (a: U, s: T) => V): Lens",
  },
  "lensIndex": Object {
    "typing": "lensIndex(n: number): Lens",
  },
  "lensProp": Object {
    "typing": "lensProp(str: string): {
  <T, U>(obj: T): U",
  },
  "map": Object {
    "typing": "map<T, U>(mapFn: MapFunctionObject<T, U>, x: Dictionary<T>): Dictionary<U>",
  },
  "match": Object {
    "typing": "match(regexp: RegExp, str: string): any[]",
  },
  "max": Object {
    "typing": "max<T extends Ord>(a: T, b: T): T",
  },
  "maxBy": Object {
    "typing": "maxBy<T>(keyFn: (a: T) => Ord, a: T, b: T): T",
  },
  "mean": Object {
    "typing": "mean(list: ReadonlyArray<number>): number",
  },
  "median": Object {
    "typing": "median(list: ReadonlyArray<number>): number",
  },
  "merge": Object {
    "typing": "merge<T1, T2>(a: T1, b: T2): Merge<T2, T1>",
  },
  "min": Object {
    "typing": "min<T extends Ord>(a: T, b: T): T",
  },
  "minBy": Object {
    "typing": "minBy<T>(keyFn: (a: T) => Ord, a: T, b: T): T",
  },
  "modulo": Object {
    "typing": "modulo(a: number, b: number): number",
  },
  "multiply": Object {
    "typing": "multiply(a: number, b: number): number",
  },
  "negate": Object {
    "typing": "negate(a: number): number",
  },
  "none": Object {
    "typing": "none<T>(fn: (a: T) => boolean, list: ReadonlyArray<T>): boolean",
  },
  "not": Object {
    "typing": "not(x: any): boolean",
  },
  "nth": Object {
    "typing": "nth<T>(n: number, list: ReadonlyArray<T>): T | undefined",
  },
  "omit": Object {
    "typing": "omit<T>(propsToOmit: string | string[], obj: Dictionary<T>): Dictionary<T>",
  },
  "over": Object {
    "typing": "over<T>(lens: Lens, fn: Arity1Fn, value: T): T",
  },
  "partial": Object {
    "typing": "partial<V0, V1, T>(fn: (x0: V0, x1: V1) => T, x0: V0): (x1: V1) => T",
  },
  "path": Object {
    "typing": "path<Input, T>(pathToSearch: string | string[], obj: Input): T | undefined",
  },
  "pathOr": Object {
    "typing": "pathOr<T>(defaultValue: T, pathToSearch: Path, obj: any): T",
  },
  "paths": Object {
    "typing": "paths<Input, T>(pathsToSearch: (string | string[])[], obj: Input): (T | undefined)[]",
  },
  "pick": Object {
    "typing": "pick<T>(propsToPick: string | string[], obj: Dictionary<T>): Dictionary<T>",
  },
  "pickAll": Object {
    "typing": "pickAll<T, U>(names: ReadonlyArray<string>, obj: T): U",
  },
  "pipe": Object {
    "typing": "pipe<T1>(fn0: () => T1): () => T1",
  },
  "pluck": Object {
    "typing": "pluck<T>(property: number, arr: ReadonlyArray<T>): T",
  },
  "prepend": Object {
    "typing": "prepend<T>(x: T, arr: ReadonlyArray<T>): T[]",
  },
  "product": Object {
    "typing": "product(list: ReadonlyArray<number>): number",
  },
  "prop": Object {
    "typing": "prop<P extends keyof T, T>(propToFind: P, obj: T): T[P]",
  },
  "propEq": Object {
    "typing": "propEq<T>(propToFind: string | number, valueToMatch: T, obj: any): boolean",
  },
  "propIs": Object {
    "typing": "propIs<P extends keyof T, T>(type: any, name: P, obj: T): boolean",
  },
  "propOr": Object {
    "typing": "propOr<T, U, V>(val: T, p: string, obj: U): V",
  },
  "range": Object {
    "typing": "range(start: number, end: number): number[]",
  },
  "reduce": Object {
    "typing": "reduce<T, TResult>(fn: (acc: TResult, elem: T, i: number) => TResult, acc: TResult, list: ReadonlyArray<T>): TResult",
  },
  "reject": Object {
    "typing": "reject<T>(filterFn: FilterFunctionArray<T>): (x: T[]) => T[]",
  },
  "repeat": Object {
    "typing": "repeat<T>(a: T, n: number): T[]",
  },
  "replace": Object {
    "typing": "replace(strOrRegex: RegExp | string, replacer: string, str: string): string",
  },
  "reverse": Object {
    "typing": "reverse<T>(list: ReadonlyArray<T>): T[]",
  },
  "set": Object {
    "typing": "set<T, U>(lens: Lens, a: U, obj: T): T",
  },
  "slice": Object {
    "typing": "slice(a: number, b: number, list: string): string",
  },
  "sort": Object {
    "typing": "sort<T>(sortFn: (a: T, b: T) => number, arr: ReadonlyArray<T>): T[]",
  },
  "sortBy": Object {
    "typing": "sortBy<T>(sortFn: (a: T) => Ord, arr: ReadonlyArray<T>): T[]",
  },
  "split": Object {
    "typing": "split(sep: string | RegExp): (str: string) => string[]",
  },
  "splitEvery": Object {
    "typing": "splitEvery<T>(a: number, list: ReadonlyArray<T>): T[][]",
  },
  "startsWith": Object {
    "typing": "startsWith(a: string, list: string): boolean",
  },
  "subtract": Object {
    "typing": "subtract(a: number, b: number): number",
  },
  "sum": Object {
    "typing": "sum(list: ReadonlyArray<number>): number",
  },
  "symmetricDifference": Object {
    "typing": "symmetricDifference<T>(list1: ReadonlyArray<T>, list2: ReadonlyArray<T>): T[]",
  },
  "tail": Object {
    "typing": "tail<T>(arrOrStr: ReadonlyArray<T>): T[]",
  },
  "take": Object {
    "typing": "take<T>(num: number, arrOrStr: ReadonlyArray<T>): T[]",
  },
  "takeLast": Object {
    "typing": "takeLast<T>(num: number, arrOrStr: ReadonlyArray<T>): T[]",
  },
  "tap": Object {
    "typing": "tap<T>(fn: (a: T) => any, value: T): T",
  },
  "test": Object {
    "typing": "test(regExpression: RegExp): (str: string) => boolean",
  },
  "times": Object {
    "typing": "times<T>(fn: (i: number) => T, n: number): T[]",
  },
  "toLower": Object {
    "typing": "toLower(str: string): string",
  },
  "toPairs": Object {
    "typing": "toPairs<S>(obj: { [k: string]: S } | { [k: number]: S }): [string, S][]",
  },
  "toString": Object {
    "example": [Function],
    "explanation": [Function],
    "failedRamdaSpecs": [Function],
    "note": [Function],
    "typing": "toString<T>(val: T): string",
  },
  "toUpper": Object {
    "typing": "toUpper(str: string): string",
  },
  "transpose": Object {
    "typing": "transpose<T>(list: T[][]): T[][]",
  },
  "trim": Object {
    "typing": "trim(str: string): string",
  },
  "type": Object {
    "typing": "type(val: any): \\"Object\\" | \\"Number\\" | \\"Boolean\\" | \\"String\\" | \\"Null\\" | \\"Array\\" | \\"Function\\" | \\"Undefined\\" | \\"Async\\" | \\"Promise\\" | \\"RegExp\\" | \\"NaN\\"",
  },
  "uniq": Object {
    "typing": "uniq<T>(arr: ReadonlyArray<T>): T[]",
  },
  "uniqWith": Object {
    "typing": "uniqWith<T, U>(fn: (x: T, y: T) => boolean, arr: ReadonlyArray<T>): T[]",
  },
  "update": Object {
    "typing": "update<T>(index: number, value: T, list: ReadonlyArray<T>): T[]",
  },
  "values": Object {
    "typing": "values<T extends object, K extends keyof T>(obj: T): T[K][]",
  },
  "view": Object {
    "typing": "view<T, U>(lens: Lens): (obj: T) => U",
  },
  "without": Object {
    "typing": "without<T>(list1: ReadonlyArray<T>, list2: ReadonlyArray<T>): T[]",
  },
  "xor": Object {
    "typing": "xor(a: boolean, b: boolean): boolean",
  },
  "zip": Object {
    "typing": "zip<K, V>(list1: ReadonlyArray<K>, list2: ReadonlyArray<V>): KeyValuePair<K, V>[]",
  },
  "zipObj": Object {
    "typing": "zipObj<T>(keys: ReadonlyArray<string>, values: ReadonlyArray<T>): { [index: string]: T }",
  },
}
`;
